layer/layer_store.go

```diff
@@ -446,12 +453,14 @@ func (ls *layerStore) releaseLayer(l *roLayer) ([]Metadata, error) {
 		if l.hasReferences() {
 			panic("cannot delete referenced layer")
 		}
+		// Remove layer from layer map first so it is not considered to exist
+		// when if ls.deleteLayer fails.
+		delete(ls.layerMap, l.chainID)
```

Comment:
Why is this?

Reply:
To avoid situations which I explained on https://github.com/moby/moby/issues/38488#issuecomment-492801678 when layer is removed from layer map it will be re-downloaded if any new containers will be deployed which is based on same (original) layer.